### DISCUSSION TITLE
[code] coding standard - code checker - code formatter

### DISCUSSION THREAD

**Discussion by artesim-zz** [Ortho4XP/Ortho4XP#25] [Ideas]
This might be too much for most people, but I think @girotobial and me are on the
same page on this subject.

There are many advantages to follow the same coding standard :
- lower potential of false merge conflicts
- both the check and the formatting can be automated by IDEs or external tools
- newcomers who already know python from other projects are more likely to feel
  at home here, and contribute more easily.
  
On this subject, @girotobial mentioned the black formatter [in this comment](https://github.com/oscarpilote/Ortho4XP/issues/165#issuecomment-956329969).
I'd be interested to learn more about that, maybe you can help me set that up in
PyCharm ? (by the way, is this what you use ?).

**Comment by girotobial** [comment]
I use VsCode for my IDE, but you can easily use black from the command line. I expect PyCharm will have automatic formatting options available, as far as I know it is an excellent IDE choice. 

Over the next few days I'm going to drop in a PR for a suggestion for the code formatting guidlines. Including

- Formatting,
- Linting/Error checks
- Auto import sort. (Although this may have to wait until we sort out the config. In my testing I found that the order of imports matter a lot in the code currently)
- Commit hooks to enforce style.
- CI checks
